src/app/donation-complete/donation-complete.component.ts

```diff
@@ -283,7 +283,7 @@ export class DonationCompleteComponent implements OnInit {
 
   private setSocialShares(campaign: Campaign) {
     const prefix = environment.donateGlobalUriPrefix;
-    this.shareUrl = `${prefix}/campaign/${campaign.id}`;
+    this.shareUrl = encodeURIComponent(`${prefix}/campaign/${campaign.id}`);
```

Comment:
ah you're quite right, it should be `encodeURIComponent` not `encodeURI`. Look at the docs on MDN I don't really understand now what the point of encodeURI is or when anyone would want that - encoding always has to be based on the output context, this is correct in the context we have that it's going in a URI. I don't know what context encodeURI is supposed to be used in.

Reply:
Not a blocker but might be worth renaming the property to `this.encodedShareUrl` to make sure we don't double encode it and make sure we don't forget to encode it if we assign it from anywhere else. Effectively it's a different type.

Reply:
Looking at https://stackoverflow.com/a/9247370/2526181 `encodeURI` seems to be a useless function.